libraries/botbuilder/src/teamsInfo.ts

```diff
@@ -155,13 +159,33 @@ export class TeamsInfo {
             },
             activity: activity,
         } as ConversationParameters;
-        const connectorClient = (context.adapter as BotFrameworkAdapter).createConnectorClient(
-            context.activity.serviceUrl
-        );
-        const conversationResourceResponse = await connectorClient.conversations.createConversation(convoParams);
-        const conversationReference = TurnContext.getConversationReference(context.activity);
-        conversationReference.conversation.id = conversationResourceResponse.id;
-        return [conversationReference as ConversationReference, conversationResourceResponse.activityId];
+
+        let conversationReference: Partial<ConversationReference>;
+        let newActivityId: string;
+
+        if (botAppId && context.adapter instanceof CloudAdapter) {
+            await context.adapter.createConversationAsync(
+                botAppId,
+                Channels.Msteams,
+                context.activity.serviceUrl,
+                null,
+                convoParams,
+                async (t: TurnContext): Promise<void> => {
```

Comment:
The types here should all be inferred.

Reply:
Yep. Looks like they are. To clarify, @joshgummersall, I should remove `: TurnContext: Promise<void>`?

Reply:
yep, I prefer to omit anything that can be inferred by the compiler. so this could be
```typescript
...,
(t) => {
  ...
}
```